sklego/decomposition/pca_reconstruction.py

```diff
@@ -95,6 +97,7 @@ def fit(self, X, y=None):
             If `threshold` is `None`.
         """
         X = check_array(X, estimator=self, dtype=FLOAT_DTYPES)
+        _check_n_features(self, X, reset=True)
```

Comment:
you probably want to call `validate_data` instead of `check_array`, `check_X_y`, and `_check_n_features`. In sklearn itself, we rarely call these functions. It's almost always `validate_data`

Reply:
Thanks @adrinjalali ! I will have a deeper look.

If you look at the first 7 commits of this PR, that was exactly how I approached it - i.e. moving from `check_array` and `check_X_y` to `validate_data`. However, many tests were failing (especially for meta estimators), thus I move forward with the safe path of adjusting what we already have.

Reply:
I'd be happy to have a look if you point me to some errors.

Reply:
It seems it was a skill issue of mine! Should be fixed now 😇